### PR TITLE
Add libs directory

### PR DESCRIPTION
**Pull request by dschatzberg**
This PR adds a new libs directory and the filesystem ebb library.

### CODE REVIEW

File: libs/filesystem/cmake/FindCapnp.cmake [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+#   target_link_library(a ${CAPNP_LIBRARIES})
+#
+# For out-of-source builds:
+#
+#   set(CAPNPC_OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR})
+#   include_directories(${CAPNPC_OUTPUT_DIR})
+#   capnp_generate_cpp(...)
+#
+
+# CAPNP_GENERATE_CPP ===========================================================
```
Comment (handong32) [inline libs/filesystem/cmake/FindCapnp.cmake]: Is the commented section required?

Comment (dschatzberg) [inline libs/filesystem/cmake/FindCapnp.cmake]: No, it's not required.

Comment (handong32) [inline libs/filesystem/cmake/FindCapnp.cmake]: Are you including it in the commit or should you delete them

Comment (dschatzberg) [inline libs/filesystem/cmake/FindCapnp.cmake]: Why would I delete documentation? Regardless, this is just a file I copied from our repo. I don't intend to modify it

**Comment by handong32** [comment]
@BetsyRT r+

**Comment by dschatzberg** [comment]
@BetsyRT r+

**Comment by BetsyRT** [comment]
:pushpin: Commit 1208897 has been approved by `dschatzberg`

**Comment by BetsyRT** [comment]
:hourglass: Testing commit 1208897 with merge c6c3b9d...

**Comment by BetsyRT** [comment]
:sunny: Test successful - [runtests](http://192.168.73.2:8010/builders/runtests/builds/90)
Approved by=dschatzberg
Pushing c6c3b9d12632cf7611fe43ca856aaaece6c6d532 to master...

**Pull request merged**